src/data_analysis/symmetry_checker.py

```diff
@@ -0,0 +1,239 @@
+                  (h_dummy,  ""),
+                  (h_valid,  "valid"),
+                  (h_nans,   "nans"),
+                  (h_zeros,  "zeros"),
+                  (h_morton, "morton")
+                  )
+
+        leg = ax.legend(*zip(*legend),
+                        title="Difference:",
+                        loc="upper left",
+                        bbox_to_anchor=(1.05, 1.02))
+        ax.set_axisbelow(True)
+        plt.tight_layout()
+        leg._legend_box.align = "left"
+        for lh in leg.legendHandles:
+            lh.set_alpha(1)
+
+        if self._out_dir:
+            filename = ("bland-altman-%s-%s.png" % (col.lower(), pid))
+            path = self._out_dir / "plots" / filename
+            save_figure(path=path, fig=fig)
+        plt.close(fig)
+
+        info = pd.DataFrame(columns=x.columns, dtype=float)
+        info.loc["counts"] = n
+        # null: {NaN, None NaT}
+        info.loc["nans"] = nans.sum(axis=0)
+        info.loc["nan_ratio"] = info.loc["nans"] / n
+        info.loc["zero_ratio"] = (x == 0).sum(axis=0) / n
+        info.loc["const_ratio"] = ((x.shift(1)-x) == 0).sum(axis=0) / (n-1)
+
+        info_diff = pd.Series(name="diff", dtype=float)
+        info_diff["mean"] = diff_mean
+        info_diff["std"] = diff_std
+        info_diff["5%"]  = diff.quantile(0.05)
+        info_diff["25%"] = diff.quantile(0.25)
+        info_diff["50%"] = diff.quantile(0.50)
+        info_diff["75%"] = diff.quantile(0.75)
+        info_diff["95%"] = diff.quantile(0.95)
+        info_diff["corr"] = x["left"].corr(x["right"])
+
+        # Format output as series with multi-level index.
+        ret = pd.concat([info.unstack(),
+                         info_diff.to_frame().unstack()],
+                        axis=0)
+        ret.name = (pid, col)
+
+        return ret
+
+
+    def run(self):
+        file_pattern = re.compile(r"([0-9]*)(L|R).*")
+        files = sorted(self._data_dir.glob("*.csv"))
+        files = {file_pattern.match(f.stem).groups():f for f in files}
+        files = pd.Series(files)
+        rets = []
+        if files.empty:
+            warnings.warn("No files found under the following location:\n%s" % self._data_dir)
+            return
+        for key, df in files.groupby(level=0):
+            if len(df)!=2:
+                warnings.warn("No matching files found for patient '%s'" % key)
+                continue
+            file_left = df[(key, "L")]
+            file_right = df[(key, "R")]
+            df = self._load_data(file_left=file_left, file_right=file_right)
+            print("processing data for patient %s ..." % key)
+            for col in self._coi:
+                ret = self._analyze_per_patient(df, col=col, pid=key)
+                rets.append(ret)
+
+        rets = pd.concat(rets, axis=1)
+        rets.to_csv(self._out_dir / "results.csv")
+        #rets = pd.read_csv(self._out_dir / "results.csv", header=[0,1], index_col=[0,1])
```

Comment:
I try not to have uncommented code since in the future people will not remember why it was uncommented

Reply:
Sometimes it's nice to see what has been tried before. But I agree, for production code this is bad practice.